### PR TITLE
refactor: Add header panel types

### LINKED ISSUE #13594: Add TypeScript types to HeaderPanel

### PR DESCRIPTION
**Pull request by danoro96**
Closes #13594

Typed out HeaderPanel.

#### Changelog

**New**

- Added myself to contributors

**Changed**

- Changed HeaderPanel.js to HeaderPanel.tsx and added types as well as fixing an error

#### Testing / Reviewing

Run in storybook and check that there are no issues.

### CODE REVIEW

**Comment by danoro96** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/react/src/components/UIShell/HeaderPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+import { usePrefix } from '../../internal/usePrefix';
+import { keys, match } from '../../internal/keyboard';
+import { useWindowEvent } from '../../internal/useEvent';
+import { useMergedRefs } from '../../internal/useMergedRefs';
+
+interface HeaderPanelProps {
+  addFocusListeners: boolean;
+  children: ReactNode;
+  expanded: boolean;
+  href: string;
```
Comment (tay1orjones) [inline packages/react/src/components/UIShell/HeaderPanel.tsx]: I don't think any of these are required

**Review by tay1orjones — CHANGES_REQUESTED** [review]
Thank you! This looks good, just one minor piece on required vs. optional on a few props

**Review by tay1orjones — APPROVED** [review]

**Review by andreancardona — APPROVED** [review]

File: packages/react/src/components/UIShell/HeaderPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+    const controlled = useRef(expanded !== undefined).current;
+    const [expandedState, setExpandedState] = useState(expanded);
+    const expandedProp = controlled ? expanded : expandedState;
+
+    const [lastClickedElement, setLastClickedElement] = useState(null);
+
+    const className = cx(`${prefix}--header-panel`, {
+      [`${prefix}--header-panel--expanded`]: expandedProp,
+      [customClassName]: !!customClassName,
```
Comment (tay1orjones) [inline packages/react/src/components/UIShell/HeaderPanel.tsx]: It looks like CI is failing on this line

**Review by tay1orjones — APPROVED** [review]
Fixed, looks good 👍

**Pull request merged**